pkg/api/plugins.go

```diff
@@ -492,18 +491,6 @@ func (hs *HTTPServer) UninstallPlugin(c *contextmodel.ReqContext) response.Respo
 }
 
 func translatePluginRequestErrorToAPIError(err error) response.Response {
-	if errors.Is(err, backendplugin.ErrPluginNotRegistered) {
-		return response.Error(http.StatusNotFound, "Plugin not found", err)
-	}
-
-	if errors.Is(err, backendplugin.ErrMethodNotImplemented) {
-		return response.Error(http.StatusNotFound, "Not found", err)
-	}
-
-	if errors.Is(err, backendplugin.ErrPluginUnavailable) {
-		return response.Error(http.StatusServiceUnavailable, "Plugin unavailable", err)
-	}
```

Comment:
Here 503 service unavailable and in plugins/errors.go 500 internal server error. Need to decide which one it should be.

Reply:
In this case, I think a 503 is not the correct code? [According to the RFC](https://www.rfc-editor.org/rfc/rfc9110.html#name-503-service-unavailable) 503 errors are temporary, this sounds more like a 500 to me. Any issue changing it?

Reply:
No issues with it. However, could be discussed if it should be denoted as a `downstream` error or not. If a backend plugin has crashed then it would be a downstream error - nothing a Grafana operator should be paged about. If plugin is not installed it feels like a user error so maybe bad request would be more suitable - tricky. @wbrowne any input?

Maybe follow up PR to change this?

Reply:
not related to this PR but a downstream error is marked as "Internal" so it will return a 500, is that correct?
https://github.com/grafana/grafana/blob/main/pkg/plugins/errors.go#L15

Reply:
Can be discussed, but kept it like that when working on #68480 to make it behave as earlier. Might be something we want to revisit eventually in the future